### ISSUE TITLE
Broken builds and lack of free CI/CD with travis-ci

### ISSUE THREAD

**Issue by avmarchenko** [exa-analytics/exa#220]
**Describe the bug**
Builds need to be updated - they are currently broken (travis - linux/osx, and appveyor - windows). Our thinking is that version pinning/updates to code to account for changes in dependencies need to be made.

**Comment by tjduigna** [comment]
This might be a good opportunity to unify build dependency specification and source that in the ci pipelines. Also, we might want to rethink the use of conda for build environments as other managers (like pipenv or poetry) feel a bit more light weight.

**Comment by avmarchenko** [comment]
+1 for using something other than conda - but I'd advocate we don't lose the ability to publish packages to pypi.org and anaconda.org.

**Comment by tjduigna** [comment]
Right, publishing conda packages definitely needs to be maintained. It would be nice to have a strategy that allows testing pipeline logic normally reserved for releases, e.g. publishing releases tagged as staging on a less restricted subset of pipeline parameters.

**Comment by avmarchenko** [comment]
Evaluating some of the free tier ci systems out there leads me to believe we need to pivot to GitHub actions or circle-ci. Those two options appear to have a persistent free tier. Other options (travis, appveyor, etc.) have a absolute limit on the amount of free tier compute.

**Comment by avmarchenko** [comment]
Since I'm not familiar with github actions I built a little expt https://github.com/avmarchenko/test-actions
